Userland/Libraries/LibWeb/DOM/Element.cpp

```diff
@@ -2373,7 +2373,9 @@ Optional<Element::Directionality> Element::auto_directionality() const
 
     // 1. If element is an auto-directionality form-associated element:
     if (is_auto_directionality_form_associated_element()) {
-        auto const& value = static_cast<HTML::HTMLInputElement const&>(*this).value();
+        auto const* form_associated_element = dynamic_cast<HTML::FormAssociatedElement const*>(this);
+        VERIFY(form_associated_element);
```

Comment:
Shouldn't this be what `verify_cast` does?

Reply:
No, `verify_cast` uses `static_cast`, and we cannot static cast from `Element` to `FormAssociatedElement` as there is no direct inheritance chain there. We have to use only `dynamic_cast`.